src/npcmove.cpp

```diff
@@ -346,27 +346,17 @@ void npc::execute_action( npc_action action )
         break;
 
     case npc_reload: {
-        // fetch potential ammo from npc possessions only and exclude empty magazines
-        auto ammo = find_ammo( weapon, false, -1 );
-        ammo.erase( std::remove_if( ammo.begin(), ammo.end(), [this]( const item_location& e ) {
-            return weapon.can_reload( e->typeId() );
-        } ), ammo.end() );
+        auto usable_ammo = find_usable_ammo();
 
-        // prefer either most full magazine or most plentiful ammo stack
-        std::sort( ammo.begin(), ammo.end(), []( const item_location& lhs, const item_location& rhs ) {
-            return rhs->ammo_remaining() < lhs->ammo_remaining();
-        } );
-
-        // @todo handle reloading of empty magazines
-
-        if( ammo.empty() ) {
+        if( !usable_ammo ) {
             debugmsg( "npc_reload failed: no usable ammo" );
             break;
         }
 
-        int reload_time = item_reload_cost( weapon, *ammo[0] );
+        int reload_time = item_reload_cost( weapon, *usable_ammo );
 
-        if( !weapon.reload( *this, std::move( ammo[0] ) ) ) {
```

Comment:
I think so. After I removed the default argument, I didn't need to change anything else.